sentry-ruby/lib/sentry/envelope.rb

```diff
@@ -7,19 +7,20 @@ class Item
       STACKTRACE_FRAME_LIMIT_ON_OVERSIZED_PAYLOAD = 500
       MAX_SERIALIZED_PAYLOAD_SIZE = 1024 * 1000
 
-      attr_accessor :headers, :payload
+      attr_accessor :headers, :payload, :is_json
 
-      def initialize(headers, payload)
+      def initialize(headers, payload, is_json: true)
         @headers = headers
         @payload = payload
+        @is_json = is_json
       end
 
       def type
         @headers[:type] || 'event'
       end
 
       def to_s
-        [JSON.generate(@headers), JSON.generate(@payload)].join("\n")
+        [JSON.generate(@headers), @is_json ? JSON.generate(@payload) : @payload].join("\n")
```

Comment:
Before this PR, every item's `@payload` is a hash. And if I understand it correctly, metrics payload would be strings instead (as we call `bytesize` on them)? If that's the case, maybe we can use `@payload` type as the condition here and don't need the `is_json` attribute for `Item`? Like:

```rb
[JSON.generate(@headers), @payload.is_a?(String) ? @payload : JSON.generate(@payload)].join("\n")
```